test/commons/color/get-text-shadow-colors.js

```diff
@@ -192,4 +192,154 @@ describe('axe.commons.color.getTextShadowColors', function () {
     assert.equal(shadowColors[0].blue, 0);
     assert.equal(shadowColors[0].alpha, 0);
   });
+
+  describe('with shadows that combine to a stroke', () => {
+    const opt = { minRatio: 0.01 };
+    it('combines multiple shadows without blur into a single stroke', () => {
+      fixture.innerHTML = `
+        <span style="text-shadow:
+          0 -2px #F00,
+          2px 0 #F00,
+          0 2px #F00,
+          -2px 0 #F00;
+        ">Hello wold</span>
+      `;
+      const shadowColors = getTextShadowColors(fixture.firstElementChild, opt);
+      assert.deepEqual(shadowColors, [
+        {
+          red: 255,
+          green: 0,
+          blue: 0,
+          alpha: 1
+        }
+      ]);
+    });
+
+    it('only combines shadows thinner than minRatio', () => {
+      const minRatio = 0.1; // .1em (has to be greater than 1px, or this will be null)
+      fixture.innerHTML = `
+        <span style="text-shadow:
+          0 -2px ${minRatio}em #000,
+          2px 0 ${minRatio}em #000,
+          0 2px ${minRatio}em #000,
+          -2px 0 ${minRatio}em #000;
+        ">Hello wold</span>
+      `;
+      const shadowColors = getTextShadowColors(fixture.firstElementChild, {
+        minRatio
+      });
+      assert.lengthOf(shadowColors, 4);
+    });
+
+    it('skips raised-letter effects (shadows on 1 or 2 sides)', () => {
```

Comment:
Would be good to have a test that verifies that exactly 3 sides is sufficient to be not-skipped